src/S3Bucket/Components/DownloadButton.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <VBtn icon="mdi-file-download" :disabled="isDisabled" size="x-large" title="Download" @click="downloadFiles"/>
+</template>
+    
+<script setup>
+import { computed } from "vue";
+import { convertObjectToFormData } from '../../helpers/axiosHelper';
+/**
+ * @typedef {Object} Props
+ * @property {ReturnType<import("@cloudbolt/js-sdk").createApi>} Props.api - The authenticated API instance
+ * @property {String} Props.resourceId - The S3 Bucket resource Id
+ * @property {String} Props.location - The S3 Bucket location
+ * @property {Array} Props.selectedItems - The selected S3 Bucket items
+ */
+/** @type {Props} */
+const props = defineProps({
+  api: {
+    type: Object,
+    required: true,
+  },
+  resourceId: {
+    type: String,
+    required: true,
+  },
+  location: {
+    type: String,
+    required: true,
+  },
+  selectedItems: {
+    type: Array,
+    default: () => [],
+  },
+});
+// Disabled download if there are no items, or folder items
+const isDisabled = computed(() => {
+  if (props.selectedItems.length === 0 || props.selectedItems.findIndex((entry) => !entry.is_file) !== -1) {
+    return true
+  }
+  return false
+})
+
+const filePaths = computed(() => {
+  const allFiles = []
+  props.selectedItems.forEach((item) => {
+    allFiles.push({
+      path: item.url,
+      location: props.location
+    })
+  })
+
+  return allFiles
+})
```

Comment:
Nit: This could be simplified with a map:
```suggestion
const filePaths = computed(() => 
  props.selectedItems.map((item) => ({
    path: item.url,
    location: props.location
  }))
)
```